python/ray/data/context.py

```diff
@@ -97,6 +97,9 @@
 # If disabled, users can still manually print stats with Dataset.stats().
 DEFAULT_AUTO_LOG_STATS = False
 
+# Whether to enable optimizer.
+DEFAULT_OPTIMIZER_ENABLED = False
```

Comment:
Make this toggleable with an env var like the others?

Reply:
It's easier to experiment with on the command line (like `RAY_DATASET_NEW_EXECUTION_BACKEND=1 RAY_DATASET_USE_STREAMING_EXECUTOR=1 python demo.py`).